python/openvino_tokenizers/tokenizer_pipeline.py

```diff
@@ -839,18 +839,30 @@ def get_ov_subgraph(self, input_nodes):
 
         op_inputs = []
         input_nodes_iter = iter(input_nodes)
-        for node in self.inputs:
-            if isinstance(node, Sequence):
-                op_inputs.extend(islice(input_nodes_iter, 3))
-            elif isinstance(node, AddToken):
-                # Put a scalar as a ragged tensor with scalar shape and a single element
+
+        segment_ids = []
+        segment_index = 0
+        for key, group_iter in groupby(self.inputs, key=type):
+            if key is Sequence:
+                for sequence in group_iter:
+                    op_inputs.extend(islice(input_nodes_iter, 3))
+                segment_ids.append(self.segment_ids[segment_index])
+                segment_index += 1
+            elif key is AddToken:
+                ids = [node._token_id for node in group_iter]
+                
+                tmp_seg_ids = np.array(self.segment_ids[segment_index:segment_index + len(ids)])
+                assert (tmp_seg_ids == tmp_seg_ids[0]).all()
```

Comment:
This assert can be avoided, if you expand a key to `lambda inp: (type(inp), inp.token_id)`

Reply:
done